GraphClientFactory.md

```diff
@@ -18,7 +18,14 @@ There are two primary usage scenarios for the client factory.  Developers who wi
   - Set the SdkVersion header with the appropriate moniker based on the following structure `graph-{lang}-{version}`.  
 - Enable the developer to select a supported sovereign cloud using an enumerated list.  Selecting the sovereign cloud should ensure that the AuthenticationProvider uses the appropriate Authentication Endpoint.
 - Enable a developer to configure a HTTP proxy that will be used for outgoing requests.
-
+- Enable a developer provide custom URLs or endpoints. 
```

Comment:
`Enable a developer provide custom URLs or endpoints.`
`Enable a developer to provide custom URLs or endpoints.`

I think we should consider using the RequestContext mechanism to pass through the CustomUrls option through the middleware, in addition to an on individual request option where we can disable applying our headers.

Should this only be applicable to our default middleware or should the mechanism be accessible to custom middleware?  Ideally custom middleware should be able to take advantage of this.

Do you have an example of a workload failing with an unexpected header? I can only think of OneDrive on redirection to a download/upload URL.

Reply:
An example of a workload failing with an unexpected header - I got this error with Outlook large file upload too.